### PR TITLE
Upgrade dependencies for bundler and rake

### PR DESCRIPTION
**Pull request by ajkerr**
There are a few GitHub alerts on this gem related to vulnerable versions of bundler and rake. This PR should clean them up.

### CODE REVIEW

**Review by dougedey-shopify — COMMENTED** [review]
I can't think of any issues with this, but I will defer to @rafaelfranca

**Comment by ajkerr** [comment]
@rafaelfranca You good with me merging this?

File: smart_todo.gemspec [hunk 1]
```diff
@@ -30,8 +30,8 @@ Gem::Specification.new do |spec|
   spec.executables   = ['smart_todo']
   spec.require_paths = ["lib"]
 
-  spec.add_development_dependency("bundler", "~> 1.17")
-  spec.add_development_dependency("rake", "~> 10.0")
+  spec.add_development_dependency("bundler", ">= 2.2.17")
```
Comment (rafaelfranca) [inline smart_todo.gemspec:33]: Do we require bundler 2? I think should be sufficient
```suggestion
  spec.add_development_dependency("bundler", ">= 1.17")
```

File: smart_todo.gemspec [hunk 2]
```diff
@@ -30,8 +30,8 @@ Gem::Specification.new do |spec|
...
+  spec.add_development_dependency("rake", ">= 12.3.3")
```
Comment (rafaelfranca) [inline smart_todo.gemspec:34]:
```suggestion
  spec.add_development_dependency("rake", ">= 10.0")
```

**Review by rafaelfranca — CHANGES_REQUESTED** [review]
Let's relax the contraints so we don't force people to upgrade (although they should, but that is not a concern of this gem)

File: smart_todo.gemspec (hunk 1 above)
Comment (ajkerr) [reply smart_todo.gemspec:33]: I upgraded bundler here to resolve https://github.com/Shopify/smart_todo/security/dependabot/smart_todo.gemspec/bundler/open.

**Comment by ajkerr** [comment]
> Let's relax the contraints so we don't force people to upgrade (although they should, but that is not a concern of this gem)

Makes sense - if I just `bundle update` the vulnerable gems to update `Gemfile.lock` that might clear out the Dependabot alerts from GitHub.

**Review by rafaelfranca — APPROVED** [review]

**Pull request merged**